module,execution_time (sec),success,timestamp,error,error_short
cowidev.vax.batch.argentina,13.51,True,2023-05-29T07:00:21,,
cowidev.vax.batch.australia,,,2023-05-29T07:00:07,,
cowidev.vax.batch.austria_old,,,2023-05-29T07:00:07,,
cowidev.vax.batch.belgium,6.8,True,2023-05-29T07:00:14,,
cowidev.vax.batch.bolivia,4.48,True,2023-05-29T07:00:12,,
cowidev.vax.batch.canada,6.43,False,2023-05-29T07:02:03,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 282, in main
    Canada().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 270, in export
    df = df.pipe(self.pipeline, df_man)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 250, in pipeline
    .pipe(self.pipe_make_monotonic)[
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 230, in pipe_make_monotonic
    df = df.pipe(self.make_monotonic, max_removed_rows=self.max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
",1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
cowidev.vax.batch.chile,0.94,True,2023-05-29T07:00:08,,
cowidev.vax.batch.czechia,36.2,True,2023-05-29T07:00:43,,
cowidev.vax.batch.denmark,,,2023-05-29T07:00:07,,
cowidev.vax.batch.ecdc,91.83,True,2023-05-29T07:01:39,,
cowidev.vax.batch.ecuador,,,2023-05-29T07:00:07,,
cowidev.vax.batch.estonia,1.1,True,2023-05-29T07:00:08,,
cowidev.vax.batch.france,6.85,True,2023-05-29T07:00:14,,
cowidev.vax.batch.germany,0.15,True,2023-05-29T07:00:07,,
cowidev.vax.batch.greece,16.32,True,2023-05-29T07:00:23,,
cowidev.vax.batch.hong_kong,14.94,True,2023-05-29T07:00:22,,
cowidev.vax.batch.indonesia,,,2023-05-29T07:00:07,,
cowidev.vax.batch.ireland,8.73,True,2023-05-29T07:00:16,,
cowidev.vax.batch.israel,7.54,True,2023-05-29T07:00:15,,
cowidev.vax.batch.italy,29.65,True,2023-05-29T07:00:37,,
cowidev.vax.batch.japan,9.54,False,2023-05-29T07:02:12,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 375, in main
    Japan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 359, in export
    df = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 71, in read
    df_latest = self.read_latest().pipe(self.pipe_read_latest)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 100, in read_latest
    self._read_latest_booster(),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 211, in _read_latest_booster
    df = self._read_xlsx(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 233, in _read_xlsx
    raise ValueError(f""Unknown sheets: {sheets_unknown}"")
ValueError: Unknown sheets: {'5.8以降接種_一般接種（６回目）', '５回目_総接種回数', '６回目_総接種回数', '5.8以降接種_一般接種（３回目）', '5.8以降接種_一般接種（５回目）', '５回目_一般接種回数', '5.8以降接種_総接種回数', '６回目_一般接種回数', '5.8以降接種_一般接種（４回目）'}
","Unknown sheets: {'5.8以降接種_一般接種（６回目）', '５回目_総接種回数', '６回目_総接種回数', '5.8以降接種_一般接種（３回目）', '5.8以降接種_一般接種（５回目）', '５回目_一般接種回数', '5.8以降接種_総接種回数', '６回目_一般接種回数', '5.8以降接種_一般接種（４回目）'}"
cowidev.vax.batch.jersey,2.59,True,2023-05-29T07:00:10,,
cowidev.vax.batch.latvia,52.12,True,2023-05-29T07:00:59,,
cowidev.vax.batch.lithuania,5.56,True,2023-05-29T07:00:13,,
cowidev.vax.batch.luxembourg,1.92,True,2023-05-29T07:00:09,,
cowidev.vax.batch.malaysia,0.88,True,2023-05-29T07:00:08,,
cowidev.vax.batch.malta,1.44,True,2023-05-29T07:00:09,,
cowidev.vax.batch.netherlands,,,2023-05-29T07:00:07,,
cowidev.vax.batch.new_zealand,13.45,True,2023-05-29T07:00:21,,
cowidev.vax.batch.norway,0.35,True,2023-05-29T07:00:07,,
cowidev.vax.batch.peru,2.4,True,2023-05-29T07:00:10,,
cowidev.vax.batch.portugal,,,2023-05-29T07:00:07,,
cowidev.vax.batch.romania,30.03,True,2023-05-29T07:00:37,,
cowidev.vax.batch.saudi_arabia,5.38,True,2023-05-29T07:00:13,,
cowidev.vax.batch.singapore,4.65,True,2023-05-29T07:00:12,,
cowidev.vax.batch.slovakia,52.31,True,2023-05-29T07:00:59,,
cowidev.vax.batch.slovenia,1.78,True,2023-05-29T07:00:09,,
cowidev.vax.batch.south_korea,52.56,True,2023-05-29T07:01:00,,
cowidev.vax.batch.spc,31.6,True,2023-05-29T07:00:39,,
cowidev.vax.batch.sweden,14.64,True,2023-05-29T07:00:22,,
cowidev.vax.batch.switzerland,59.2,True,2023-05-29T07:01:07,,
cowidev.vax.batch.trinidad_and_tobago,4.5,True,2023-05-29T07:00:13,,
cowidev.vax.batch.ukraine,,,2023-05-29T07:00:08,,
cowidev.vax.batch.united_kingdom,13.52,True,2023-05-29T07:00:22,,
cowidev.vax.batch.united_states,13.8,True,2023-05-29T07:00:22,,
cowidev.vax.batch.uruguay,7.07,True,2023-05-29T07:00:16,,
cowidev.vax.batch.zimbabwe,3.16,True,2023-05-29T07:00:12,,
cowidev.vax.incremental.africacdc,,,2023-05-29T07:00:10,,
cowidev.vax.incremental.antigua_barbuda,,,2023-05-29T07:00:10,,
cowidev.vax.incremental.aruba,2.46,True,2023-05-29T07:00:12,,
cowidev.vax.incremental.austria,,,2023-05-29T07:00:10,,
cowidev.vax.incremental.azerbaijan,,,2023-05-29T07:00:10,,
cowidev.vax.incremental.bahrain,,,2023-05-29T07:00:10,,
cowidev.vax.incremental.bangladesh,49.67,True,2023-05-29T07:00:59,,
cowidev.vax.incremental.barbados,,,2023-05-29T07:00:12,,
cowidev.vax.incremental.brazil,0.89,True,2023-05-29T07:00:12,,
cowidev.vax.incremental.bulgaria,8.11,True,2023-05-29T07:00:20,,
cowidev.vax.incremental.china,20.14,True,2023-05-29T07:00:32,,
cowidev.vax.incremental.costa_rica,,,2023-05-29T07:00:12,,
cowidev.vax.incremental.croatia,0.98,True,2023-05-29T07:00:13,,
cowidev.vax.incremental.cuba,9.88,True,2023-05-29T07:00:22,,
cowidev.vax.incremental.curacao,,,2023-05-29T07:00:13,,
cowidev.vax.incremental.cyprus,3.2,True,2023-05-29T07:00:16,,
cowidev.vax.incremental.dominican_republic,14.4,False,2023-05-29T07:02:27,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 112, in main
    DominicanRepublic().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 68, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 26, in read
    metrics = self._parse_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 35, in _parse_metrics
    metric_candidates = _find_potential_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 91, in _find_potential_metrics
    h3 = _find_h3(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 81, in _find_h3
    raise ValueError(""More than one element found!"")
ValueError: More than one element found!
",More than one element found!
cowidev.vax.incremental.el_salvador,7.74,True,2023-05-29T07:00:20,,
cowidev.vax.incremental.faeroe_islands,1.41,True,2023-05-29T07:00:15,,
cowidev.vax.incremental.fiji,0.0,True,2023-05-29T07:00:14,,
cowidev.vax.incremental.finland,0.73,True,2023-05-29T07:00:15,,
cowidev.vax.incremental.gabon,,,2023-05-29T07:00:14,,
cowidev.vax.incremental.georgia,6.03,True,2023-05-29T07:00:20,,
cowidev.vax.incremental.greenland,,,2023-05-29T07:00:15,,
cowidev.vax.incremental.guatemala,,,2023-05-29T07:00:15,,
cowidev.vax.incremental.guernsey,,,2023-05-29T07:00:15,,
cowidev.vax.incremental.hungary,83.84,True,2023-05-29T07:01:38,,
cowidev.vax.incremental.iceland,2.94,True,2023-05-29T07:00:18,,
cowidev.vax.incremental.india,2.27,True,2023-05-29T07:00:17,,
cowidev.vax.incremental.iran,,,2023-05-29T07:00:16,,
cowidev.vax.incremental.isle_of_man,2.95,True,2023-05-29T07:00:19,,
cowidev.vax.incremental.jamaica,5.81,True,2023-05-29T07:00:22,,
cowidev.vax.incremental.kazakhstan,15.19,True,2023-05-29T07:00:31,,
cowidev.vax.incremental.kosovo,,,2023-05-29T07:00:17,,
cowidev.vax.incremental.kyrgyzstan,2.81,True,2023-05-29T07:00:20,,
cowidev.vax.incremental.laos,3.81,True,2023-05-29T07:00:21,,
cowidev.vax.incremental.macao,8.3,False,2023-05-29T07:02:35,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 90, in main
    Macao().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 75, in export
    data = self.read()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 23, in read
    iframe_url = self._get_iframe_url(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 31, in _get_iframe_url
    elem = driver.find_element_by_id(""ICovid19Monitor"")
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 360, in find_element_by_id
    return self.find_element(by=By.ID, value=id_)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 976, in find_element
    return self.execute(Command.FIND_ELEMENT, {
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 321, in execute
    self.error_handler.check_response(response)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py"", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=113.0.5672.126)

","Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=113.0.5672.126)
"
cowidev.vax.incremental.moldova,,,2023-05-29T07:00:20,,
cowidev.vax.incremental.monaco,84.49,True,2023-05-29T07:01:44,,
cowidev.vax.incremental.mongolia,0.87,True,2023-05-29T07:00:21,,
cowidev.vax.incremental.montenegro,0.49,True,2023-05-29T07:00:20,,
cowidev.vax.incremental.morocco,,,2023-05-29T07:00:20,,
cowidev.vax.incremental.myanmar,,,2023-05-29T07:00:20,,
cowidev.vax.incremental.nepal,,,2023-05-29T07:00:20,,
cowidev.vax.incremental.north_macedonia,,,2023-05-29T07:00:20,,
cowidev.vax.incremental.northern_cyprus,1.85,True,2023-05-29T07:00:22,,
cowidev.vax.incremental.paho,87.12,False,2023-05-29T07:04:02,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 195, in main
    PAHO().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 190, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 167, in pipeline
    df.pipe(self.pipe_check_columns)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 122, in pipe_check_columns
    raise ValueError(f""Missing column fields: {columns_missing}. Present columns are: {df.columns}"")
ValueError: Missing column fields: {'Country code', '2nd additional dose', 'Total doses', '1st additional dose', 'Single dose', 'First dose', 'Second dose', 'Country/ Territory'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')
","Missing column fields: {'Country code', '2nd additional dose', 'Total doses', '1st additional dose', 'Single dose', 'First dose', 'Second dose', 'Country/ Territory'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')"
cowidev.vax.incremental.pakistan,,,2023-05-29T07:00:21,,
cowidev.vax.incremental.philippines,,,2023-05-29T07:00:21,,
cowidev.vax.incremental.poland,1.09,True,2023-05-29T07:00:22,,
cowidev.vax.incremental.qatar,,,2023-05-29T07:00:21,,
cowidev.vax.incremental.russia,25.03,True,2023-05-29T07:00:46,,
cowidev.vax.incremental.saint_lucia,,,2023-05-29T07:00:21,,
cowidev.vax.incremental.serbia,12.68,True,2023-05-29T07:00:33,,
cowidev.vax.incremental.south_africa,,,2023-05-29T07:00:21,,
cowidev.vax.incremental.spain,8.6,True,2023-05-29T07:00:30,,
cowidev.vax.incremental.sri_lanka,,,2023-05-29T07:00:22,,
cowidev.vax.incremental.suriname,14.58,True,2023-05-29T07:00:36,,
cowidev.vax.incremental.taiwan,8.39,False,2023-05-29T07:04:11,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 184, in main
    Taiwan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 170, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 37, in read
    df = self._parse_table(url_pdf)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 76, in _parse_table
    raise ValueError(f""There are some unknown columns: {cols}"")
ValueError: There are some unknown columns: Index(['廠牌', '劑次', '4 月接種人次', '累計至 4 月接種人次'], dtype='object')
","There are some unknown columns: Index(['廠牌', '劑次', '4 月接種人次', '累計至 4 月接種人次'], dtype='object')"
cowidev.vax.incremental.thailand,7.33,True,2023-05-29T07:00:29,,
cowidev.vax.incremental.turkey,,,2023-05-29T07:00:22,,
cowidev.vax.incremental.united_arab_emirates,,,2023-05-29T07:00:22,,
cowidev.vax.incremental.vietnam,,,2023-05-29T07:00:22,,
cowidev.vax.incremental.who,33.72,True,2023-05-29T07:00:56,,
cowidev.vax.incremental.zambia,1.17,True,2023-05-29T07:00:23,,
